### PR TITLE
Add support to generate tests when generating field classes

### LINKED ISSUE #2349: Generate Tests Stubs
**What problem does this feature proposal attempt to solve?**
Add Test Stubs to the package. 

When creating mutations and queries, It'd be nice to have the option to create a test class with maybe one test function even if it's empty.

**Which possible solutions should be considered?**
We could add `--test` option to `lighthouse:query` & `lighthouse:mutation` commands

 Something like: 
```bash
php artisan lighthouse:query UserQuery --test
```
should generate the query in Graphql queries Directory as well as

```php 

<?php

namespace Tests\Feature\GraphQL\Queries;

use Tests\Testcase;

class UserQueryTest extends TestCase
{
    /** @test */
    public function can_get_user_query()
    {
        $response = $this->graphQL(/* @lang GraphQL */ '
            query User {
                user
            }
        ');

        $response->assertJson([
            'data' => [
                'user' => [],
            ],
        ]);
    }
}

```
I think this will help 'standardize' the way the package is used and tested.

### PR DESCRIPTION
**Pull request by yondifon**
- Support Creating Matching Tests in FieldGenerator Fixes #2349
- Changelog

 
- [x] Added or updated tests
- [x] Documented user facing changes
- [x] Updated CHANGELOG.md

**Changes**

**Breaking changes**

### CODE REVIEW

**Comment by yondifon** [comment]
I didn't see any tests for https://github.com/nuwave/lighthouse/blob/master/src/Console/FieldGeneratorCommand.php

I didn't know how to approach it. Any ideas ? @spawnia 

(Sorry for tagging you - I apologize for the noise)

File: src/Console/stubs/test.stub [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace DummyNamespace;
+
+use Tests\TestCase;
+
+class DummyClass extends TestCase
+{
+    /** @test */
+    public function it_can_use_dummy_field(): void
```
Comment (spawnia) [inline src/Console/stubs/test.stub:10]:
```suggestion
    public function test_dummy_field(): void
```

File: src/Console/stubs/test.stub [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace DummyNamespace;
+
+use Tests\TestCase;
+
+class DummyClass extends TestCase
```
Comment (spawnia) [inline src/Console/stubs/test.stub:7]:
```suggestion
final class DummyClass extends TestCase
```

File: src/Console/FieldGeneratorCommand.php [hunk 3]
```diff
@@ -22,4 +26,42 @@ protected function getOptions(): array
...
+    {
+        if (! $this->option('test') && ! $this->option('pest')) {
+            return false;
+        }
+
+        $stubPath = __DIR__.'/stubs/test.stub';
+        $stub = $this->files->get($stubPath);
+
+        $name = Str::of($path)->after($this->laravel['path'])->beforeLast('.php');
+        $fielName = $name->afterLast('/');
```
Comment (spawnia) [inline src/Console/FieldGeneratorCommand.php:40]: typo

File: src/Console/stubs/test.stub [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+{
+    /** @test */
+    public function it_can_use_dummy_field(): void
+    {
+        $response = $this->graphQL(/** @lang GraphQL */ '
+            dummyType DummyField {
```
Comment (spawnia) [inline src/Console/stubs/test.stub:13]:
```suggestion
            maybeMutation{
```

Comment (spawnia) [reply src/Console/stubs/test.stub:13]: Let's replace this with either `'mutation '` or `''` for queries.

File: src/Console/stubs/test.stub [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+                dummyField
+            }
+        ');
+
+        $response->assertJson([
+            'data' => [
+                'dummyField' => null
+            ],
+        ]);
```
Comment (spawnia) [inline src/Console/stubs/test.stub:22]:
```suggestion
        ')->assertJson([
            'data' => [
                // TODO define expected data
                'dummyField' => null,
            ],
        ]);
```

File: src/Console/FieldGeneratorCommand.php [hunk 6]
```diff
@@ -22,4 +26,42 @@ protected function getOptions(): array
             ['full', 'F', InputOption::VALUE_NONE, 'Include the seldom needed resolver arguments $context and $resolveInfo'],
         ];
     }
+
+    protected function handleTestCreation($path)
+    {
+        if (! $this->option('test') && ! $this->option('pest')) {
```
Comment (spawnia) [inline src/Console/FieldGeneratorCommand.php:32]: This only works with `'test'`, we should either implement `pest` support or warn that it does not work.

**Review by spawnia — CHANGES_REQUESTED** [review]
Thanks for adding this. There are no tests for this command, but feel free to add some.

**Comment by yondifon** [comment]
image

Couldn't figure out tests.
Sounds Horrible.  I know. 

There are no tests in the Laravel Framework Repo on Generator Class.

**Review by spawnia — APPROVED** [review]
@yondifon please re-review the changes I made.

File: src/Console/FieldGeneratorCommand.php [hunk 7]
```diff
@@ -22,4 +27,64 @@ protected function getOptions(): array
...
+        // e.g. Tests\\Feature\\Mutations\\MyMutationTest
+        $className = $operationAndFieldName
+            ->replace('/', '\\')
+            ->prepend('Tests\\Feature')
+            ->append('Test');
+
+        $stub = $this->replaceNamespace($stub, $className->toString())
+            ->replaceClass($stub, $className->toString());
+        $stub = Str::of($stub)
+            ->replace('dummyField', $operationAndFieldName->afterLast('/')->toString())
```
[the remark this answers is not in the document]
Comment (yondifon) [reply src/Console/FieldGeneratorCommand.php:55]: Just pushed one last fix. 👍🏾

**Review by spawnia — APPROVED** [review]

**Comment by spawnia** [comment]
Thank you, released with https://github.com/nuwave/lighthouse/releases/tag/v6.14.0.

**Pull request merged**